### ISSUE TITLE
Read List Defined Compound

### ISSUE THREAD

**Issue by dmlyons2** [Chemical-Curation/chemcurator_django#17]
**As a** researcher
**I want to** be able to read all defined compounds. 
**I want the** list to come back in chunks of 100 by default. But I could modify the page size with a page variable that would be a max of 1000 defined compounds per page
**I want the** sort order of the results to be by definitive structure string, descending
**I want to** be able to paginate between pages, with links to the current page, next page, previous page, first page and last page.
**I want** a meta section that indicates how many pages and the api version
**I want to** know the total count of pages in the response
**So that I** can organize my data in the chemical database and use that as "the source of truth" for my exposure database in the chemistry domain

**Request Example:**
`GET https://api.chemreg.epa.gov/defined-compounds`

**Response Example:**
```JSON
{ 
   "meta":{ 
      "totalPages":10,
      "pageSize":100,
      "apiVersion":"1.0.0-beta"
   },
   "data":[ 
      { 
         "type":"defined-compounds",
         "id":"DTXCID...",
         "attributes":{ 
            "molfile-v3000":"as;dlkjasd;lfkjasd;lfkja;sldkfjas;ldkfjasd;lkfjasd;lfkjasdf",
            "inchikey":"alsdkfjalskdjfa;l skdjfalsdkfjlsadfjklsdfkj"
         }
      },
      { 
         "type":"defined-compounds",
         "id":"DTXCID....",
         "attributes":{ 
            "molfile-v3000":"as;dlkjasd;lfkjasd;lfkja;sldkfjas;ldkfjasd;lkfjasd;lfkjasdf",
            "inchikey":"alsdkfjalskdjfa;l skdjfalsdkfjlsadfjklsdfkj"
         }
      }
   ],
   "links":{ 
      "self":"https://api.chemreg.epa.gov/defined-compounds?page[number]=3&page[size]=10",
      "first":"https://api.chemreg.epa.gov/defined-compounds?page[number]=3&page[size]=10",
      "prev":"https://api.chemreg.epa.gov/defined-compounds?page[number]=3&page[size]=10",
      "next":"https://api.chemreg.epa.gov/defined-compounds?page[number]=3&page[size]=10",
      "last":"https://api.chemreg.epa.gov/defined-compounds?page[number]=3&page[size]=10"
   }
}
```

**Acceptance Criteria:**
Using postman or something similar, read (POST) a resource id and have it return all information for that resource and a return status 404 if the resource does not exist

**Test(s):**
Check that the service will returns a list of defined compounds with the default of 100

**Comment by debboutr** [comment]
this issue is resolved in pr #66

**Comment by debboutr** [comment]
looking into trying to incorporate the `rest_framework_json_api` package to standardize the JSON formatting here we may be stuck with just utilizing the `rest_framework` package do to the fact that we are trying to override that `id` returned in the response with the Compound models `cid`. [Here](https://github.com/django-json-api/django-rest-framework-json-api/issues/155#issuecomment-301145350) is some discussion on the problem raised in `rest_framework_json_api` issues. For now, I think I will resort back to the old method that I had setup, but work on instantiating an `attributes` dict into each object for `molefile` and `inchikey` seeing as that is how the JSON spec should have it.
